Allowing all group member to submit assignment and allowing Lecture t…

Teecloudy opened the pull request:
Samrusike  - Samantha Rusike  - 201632120140
Teecloudy  - Ashly Tafadzwa Dhani - 201632120150

Teecloudy committed:
Allowing all group member to submit assignment and allowing Lecture to Edit Assignments

Two Issues have been Fixed

1)Firstly
Only Group Admins were able to submit group assignments and the system was indicating to other members that they missed the deadline.

This Problems was solved through the following stages:

Create 4 new columns 'Member_ID1-4 Varchar(50) DEFAULT 0' in the database table 'course_group_table' (this will limit each group to max 5 members you can check the file Script.php from line 1071.)

Changing the SQL query in SubmitLab.php to allow both Group Member and Group Admin to submit assignment. (However for this feature to take effect with backward compat, the group admin might need to invite other members again into the group or else they still wont be able to submit but all the group assignments will indicate that they have been submitted by the admin. For any new group after the update it works perfect.)

Changing the Course.php allows other members to be updated once 1 member submit the assignment.

2)Secondly
The Tutor couldnt modify assignments once posted.

Adding an 'Edit' Button on the Tutor's page allows him to modify new and old assignment without any duplications. This feature afftected Courses.php from line 109.

Group
Ashly Tafadzwa Dhani 201632120150
Samantha Rusike 201632120140

spm2020spring commented:
@Teecloudy  @Samrusike 

BTW, don't forget the make atomic commit next time.

Each commit represents a **baby step** and each Pull Request represents a unique feature.
)

Hui

spm2020spring commented on Course.php:
```diff
@@ -103,11 +103,11 @@
 $resultx1 = mysqli_query($con,$sql);
     
-while($row = mysqli_fetch_assoc($resultx1)) {$group_id=$row['Course_Group_id'];}  
+while($row = mysqli_fetch_assoc($resultx1)) {$_SESSION['group_id']=$row['Course_Group_id'];}  
```
- What is $resultx1?

- `$_SESSION['group_id']=$row['Course_Group_id'];`  There should be two spaces around the assignment operator (=).  Why? Easier for eyes.

**Hui**

Teecloudy replied:
$resultx1 its a variable name.
Will take note on the spaces. Actually i think we might need a code cleaner willing to go through the entire script and cleanup unnecessary duplicate, spaces, spelling errors etc and for all new lines and adjustments we will make sure to comment them and make them neat.

spm2020spring replied:
> $resultx1 its a variable name.
> Will take note on the spaces. Actually i think we might need a code cleaner willing to go through the entire script and cleanup unnecessary duplicate, spaces, spelling errors etc and for all new lines and adjustments we will make sure to comment them and make them neat.

@Teecloudy 

Code cleaning is long waited.  Thanks for offering to do that.

Hui

Teecloudy replied:
hahah i wasn't really referring to myself to do the cleaning but if there will be no one to do it, i will go on with it. I'm sure it will benefit future editors.

**Ashly**

spm2020spring commented on Course.php:
```diff
@@ -236,6 +237,7 @@
        </div>
    <div id="menu3" class="container tab-pane"><br>
          <?php
+         $group_id=$_SESSION['group_id'];
```
This line appears on line 172 too.   Does the duplication matter?

**Hui**

Teecloudy replied:
the variable was losing value, maybe it wasn't global will check it out.

spm2020spring commented on Course.php:
```diff
@@ -293,7 +295,7 @@
 Left JOIN users_table  on users_table.Student_ID=lab_report_submissions.Student_id
 left JOIN course_group_members_table on course_group_members_table.Course_Group_id=lab_report_submissions.Course_Group_id
-where Lab_Report_ID=$lab_repo_id and lab_report_submissions.Student_id='$student_id'"); 
+where Lab_Report_ID=$lab_repo_id and lab_report_submissions.Student_id='$student_id' or lab_report_submissions.Course_Group_id='$group_id'"); 
```
Do we still need the following conditional test?

`Lab_Report_ID=$lab_repo_id and lab_report_submissions.Student_id='$student_id'`


**Hui**

spm2020spring commented on Course.php:
```diff
@@ -353,18 +355,18 @@
         . "INNER JOIN lab_reports_table on lab_reports_table.Lab_Report_ID=lab_report_submissions.Lab_Report_ID "
-        . "WHERE lab_report_submissions.Student_id='$student_id' and" 
+        . "WHERE lab_report_submissions.Student_id='$student_id' or lab_report_submissions.Course_Group_id='$group_id'and" 
         . ""
         . ""
         . ""
         . " lab_reports_table.Lab_Report_ID  in (select Lab_Report_ID from lab_report_submissions"
-         . " where  (Status='Marked' or Status='Remarking') and (Student_id=$student_id)  and Course_ID=$course_id) ORDER by Submission_ID DESC");
+         . " where  (Status='Marked' or Status='Remarking') and (Student_id=$student_id or Course_Group_id=$group_id)  and Course_ID=$course_id) ORDER by Submission_ID DESC");
```
@Teecloudy 

Cannot understand this nested SQL statement in 10 seconds.  

That means it needs a higher-level comment to describe the purpose of this statement.

**Hui**

Teecloudy replied:
> @Teecloudy
> 
> Cannot under this nested SQL statement in 10 seconds.
> 
> That means it needs a higher-level comment to describe the purpose of this statement.
> 
> **Hui**

About the 10 seconds i didnt fully understand but i added "or Course_Group_id=$group_id" to make sure all other group members can be informed once the instructor marks their assignment.

spm2020spring replied:
@Teecloudy 

Sorry.  It should be "understand" not "under".   When I said "in 10 seconds", I meant in short time.  I could not understand the SQL statement in short time without the help from comments.

**Hui**

spm2020spring commented on Courses.php:
```diff
@@ -106,27 +106,121 @@ function extend_deadline(id) {
+      ?> 
+      <form method='post'   enctype='multipart/form-data' action=''>
+         <input type='hidden' name='frm_uploadlab' value='true' required=''/>
+            <input type='hidden' name='course_id' value='<?php echo ".$id." ?>' required=''/>
+                  <input type='hidden' name='url' value='<?php echo ".$course_url." ?>' required=''/>
+           
+Dealine Date/Time
+<div class='row'> 
+<div class='col-md-7'><input type='date' id='date' name='deadlinedate' placeholder='' class='form-control' required='' value="<?php echo isset($_GET['act']) && $_GET['act']=="edit" ? $Date : ""; ?>"> </div>
+<div class='col-md-5'> <input type='time' class='form-control' name='deadlinetime' value="<?php echo isset($_GET['act']) && $_GET['act']=="edit" ? $Time : ""; ?>"> </div> 
+</div>
+
+Title
+<input type='text'  name='title' placeholder='Ttle' class='form-control' required='' value="<?php echo isset($_GET['act']) && $_GET['act']=="edit" ? $Title : ""; ?>">
+Instructions
+<textarea  name='instructions' placeholder='Assignment Instructions' class='form-control' required='' value="<?php echo isset($_GET['act']) && $_GET['act']=='edit' ? $Instructions : ''; ?>"></textarea>
+Marks
+<input type='text'  name='marks' placeholder='Marks' class='form-control' required='' value="<?php echo isset($_GET['act']) && $_GET['act']=="edit" ? $Marks : ""; ?>">
+Attachment 1
+<input type='file'  name='attachment1' placeholder='Attachment 1' class='form-control'>
+
+Attachment 2
+<input type='file' name='attachment2' placeholder='Attachment 1' class='form-control'>
+
+Attachment 3
+<input type='file'  name='attachment3' placeholder='Attachment 1' class='form-control' >
+
+
+Attachment 4
+<input type='file'  name='attachment4' placeholder='Attachment 4' class='form-control' >
+<br>
+Submission Type  <input type='radio' name='type' value='Individual' required=''> Invidual
+
+<input type='radio' name='type' value='Group' required=''> Group
+<hr>
+<input type='submit' class='btn btn-primary' value='Post Lab Assignment'><br>
+</form><br><br><br><br>
+<?php
+  }}else{
+
```
@Teecloudy 
The whole form is reloaded after clicking the **Edit** button?

**Hui**

Teecloudy replied:
The Posting New report is using a different form with the Edit report. This is because the Posting new form is processed by a script in Script.php and the Edit form is processed by a script in the same file Courses.php
So i couldnt find a better way to make them use one form when they are processed in two different files unless we are going shift the lines from Script.php to Courses.php

And in addition this form still needs some improvements because right now it cant edit attachments and the <textarea> input field failed to retrieve database entries so any suggestions are welcome as we will be also figuring a way to fix this.

spm2020spring commented on Courses.php:
```diff
@@ -146,16 +240,20 @@ function extend_deadline(id) {
             echo "<div class='col-md-7'>  <h3> Lab Report Assignment list </h3>";  
                   
  error_reporting(0);
+ if(isset($_SESSION["info_Updated"])){
```
@Teecloudy 

`info_Updated` to `info_updated`

Hui

spm2020spring commented on Courses.php:
```diff
@@ -191,6 +289,7 @@ function extend_deadline(id) {
                                    $att3=$row['Attachment_link_3'];
                                     $att4=$row['Attachment_link_4'];
                                      $id=$row['Lab_Report_ID'];
+                                     $cours_id=$row['Course_ID'];
```
@Teecloudy 

is `$course_id` better than` $cours_id` ?

**Hui**

spm2020spring commented on Courses.php:
```diff
@@ -219,10 +318,11 @@ function extend_deadline(id) {
                         . "<br>"
 
-                        . "<span class='btn-default'> &nbsp;&nbsp; $count_subs Submissions ( $count_marked Marked ) &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href='~\..\Submissions.php?id=$id&header=$header&total=$marks' onclick=''> View </a>     &nbsp;&nbsp; |&nbsp;&nbsp;         <a href='#'  onclick='extend_deadline($id)'> Extend Deadline </a>  </span>         <hr> Attachments : $full_link </span> ...
+                        . "<span class='btn-default'> &nbsp;&nbsp; $count_subs Submissions ( $count_marked Marked ) &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href='Courses.php?course=".$url."&act=edit&cid=".$id."'>Edit</a>&nbsp;&nbsp; |&nbsp;&nbsp;<a href='~\..\Submissions.php?id=$id&header=$header&total=$marks' onclick=''> View </a>     &nbsp;&nbsp; |&nbsp;&nbsp;         <a href='#'  onclick='exten ...
```
@Teecloudy 

Is it possible to break this rather long line?  Why?  A bit hard for code readers like me to comprehend.

**Hui**

Teecloudy replied:
Possible

spm2020spring commented on Script.php:
```diff
@@ -682,50 +663,48 @@ function checksize($file)
+    $instructions=mysqli_real_escape_string($con,$_POST["instructions"]);
+    $title=mysqli_real_escape_string($con,$_POST["title"]);
+    
+      $url=mysqli_real_escape_string($con,$_POST["url"]);
+    
+    
+    $deadline=$deadlinedate." ".$deadlinetime;
```
@Teecloudy 

Are these two lines properly indented?

**Hui**

Teecloudy replied:
These were part of original lines. The form from the instructor's side contains input '$deadlinedat' and '$deadlinetime' separately and the writer combined both values into one ($deadline) before inserting into the database. However this was a challenge when trying to retrieve it so the instructor can edit the report, so we split it again using 
strstr($Deadline, ' ', true);
strstr($Deadline, ' '); 
at line 120 and 121 in Courses.php

spm2020spring commented on Script.php:
```diff
@@ -682,50 +663,48 @@ function checksize($file)
+  
+  
+  
+      $targetfile = $target_dir.$_FILES['attachment1']['name'];
+        $targetfile2 = $target_dir.$_FILES['attachment2']['name'];
+          $targetfile3 = $target_dir.$_FILES['attachment3']['name'];
+           $targetfile4 = $target_dir.$_FILES['attachment4']['name'];
```
@Teecloudy 

Indentation.

Hui

spm2020spring commented on Script.php:
```diff
@@ -682,50 +663,48 @@ function checksize($file)
+  
+  
+  
+      $targetfile = $target_dir.$_FILES['attachment1']['name'];
+        $targetfile2 = $target_dir.$_FILES['attachment2']['name'];
+          $targetfile3 = $target_dir.$_FILES['attachment3']['name'];
+           $targetfile4 = $target_dir.$_FILES['attachment4']['name'];
```
@Teecloudy 

What if the instructor does not have four attachments?  I hope the program won't report **Key Not Found** error.

**Hui**

Teecloudy replied:
Since this was the original code i think it wont bring any change. The student and instructors can upload upto 4 attachachments but the database can receive minimum 0 for instructor and minimum1 for student (Max being 4).

spm2020spring commented on Script.php:
```diff
@@ -737,136 +716,142 @@ function checksize($file)
+$count=0;           
+        
 
-  //check zise
```
@Teecloudy 

**zise** should be **size**?

Size for what?

Hui

spm2020spring commented on Script.php:
```diff
@@ -737,136 +716,142 @@ function checksize($file)
-  //check zise
-    if(!checksize("attachment1"))
-  {
-      echo "20 MB is the maximum file size allowed";
```
**20MB** is too big.  Use **2MB**.   Why?  Our server is low on space.  

It would be great if we could configure this limit somewhere.  No a high priority now.


**Hui**

Teecloudy replied:
We didnt affect these lines too but as i just checked i discovered thats a mistake it was supposed to be 2MB as according to this line below:

$result=$_FILES["$file"]['size']/1024/1024;

1024kb is around 1MB

spm2020spring commented on Script.php:
```diff
@@ -737,136 +716,142 @@ function checksize($file)
-      return;
-  }
-  
-  if($_FILES["attachment1"]["error"] != 0) {
```
@Teecloudy 

When is the value for the **"error" key** set?

Hui

Teecloudy replied:
Please not that we didnt affect these lines. They are the original lines. But if there may be need for any adjustments we will look at it.

spm2020spring commented on Script.php:
```diff
@@ -1086,30 +1068,62 @@ function checksize($file)
+   
+    while($row = mysqli_fetch_assoc($resultx1)) 
+    {
+      $Group_Member=$row['Group_Member']; 
+      $Group_Member4=$row['Group_Member4'];
+      $Group_Member2=$row['Group_Member2'];
+      $Group_Member3=$row['Group_Member3'];
+      $_SESSION['Group_Member4']=$Group_Member4;
+      $_SESSION['Group_Member3']=$Group_Member3;
+      $_SESSION['Group_Member2']=$Group_Member2;
+      $_SESSION['Group_Member']=$Group_Member;
```
@Teecloudy '

Better line order.  Member, Member2, Member3, and Member4 should appear in a proper order.

**Hui**

Teecloudy replied:
Noted

Teecloudy committed:
Editing report by Instructor

In the previous update, the form for editing assignments by instructor was failing to retrieve <textarea> Now its working fine. (Line 171 Courses.php)

spm2020spring commented:
@Teecloudy 

Thanks for making the `<textarea>` stuff work.   I will check.

Hui

spm2020spring commented:
> @Teecloudy @Samrusike
> 
> Thanks guys.
> 
> At least 10 hours of hard work.
> 
> Appreciate that.
> 
> * Do you have screenshots and video clips to show that the changes work?
> * Do I need to update the lrr database on the server to make your update work, for example, adding the extra four columns?  Is it possible to make them if (these columns are) not detected?
> 
> Thanks
> 
> **Hui**
> Yes check below:
> ![16](https://user-images.githubusercontent.com/37251906/78698496-04dc5e80-7935-11ea-944a-cadeb02a2ef3.jpg)
> 
> ![15](https://user-images.githubusercontent.com/37251906/78697083-dc536500-7932-11ea-8799-e7c7edcfa95d.png)

@Teecloudy 

Two **big** improvements.   

I think your group is done with the course project after these two _additional_ improvements being checked in and tested.   

- How to deploy the update to the server?  Which files are affected? Please give me a list of files.

- **Do I need to update the database on the server** to make the new update work?  I hope I can deliver the changes in one week.

Of course, you promised you will help clean up the code - a very important quality improvement for the code.

**Hui**

@spm2020spring 

**Below Files are affected:**
1. Script.php (Database 'username' and 'password' to be updated on line 13).
2. Course.php
3. Courses.php
4. SubmitLab.php

For the database, update the table 'course_groups_table'. Please refer to the lrr(1).sql file or check the screenshot below:
![asf](https://user-images.githubusercontent.com/37251906/78841327-ab0c8f00-7a2f-11ea-8ef0-e8e9c319bda9.JPG)
![tempsnip](https://user-images.githubusercontent.com/37251906/78841032-eb1f4200-7a2e-11ea-8510-0292d8555636.png)

**Ashly**

spm2020spring commented:
@Teecloudy 

Hi Ashly

Thanks for the new feature that allows a lecturer to edit assignment information.


However, it failed a regression test, reported by Zhan Shencheng (1404325791 at qq.com).  Now I am unable to post a new assignment.  See the attached picture for my input.


Got the following error in the web browser.  **Failed really badly.**


`44440 File(s) uploadedError: INSERT INTO `lab_reports_table`(`Course_ID`, `Posted_Date`, `Deadline`, `Instructions`, `Title`, `Attachment_link_1`, `Attachment_link_2`, `Attachment_link_3`, `Attachment_link_4`,Marks,Type) VALUES ('.26.','
2020-04-13 20:35','2020-04-30 23:59','None.','Quiz 5','','','','',2,'Individual')`



**Hui**


![PostNewAssignment](https://user-images.githubusercontent.com/62275785/79175308-f6f47500-7e2f-11ea-8c9a-855b35c30896.jpg)

![ErrorSQLStatement](https://user-images.githubusercontent.com/62275785/79175334-0673be00-7e30-11ea-89b7-ad6a8a5ec2da.jpg)

Teecloudy commented:
> @Teecloudy
> 
> Hi Ashly
> 
> Thanks for the new feature that allows a lecturer to edit assignment information.
> 
> However, it failed a regression test, reported by Zhan Shencheng (1404325791 at qq.com). Now I am unable to post a new assignment. See the attached picture for my input.
> 
> Got the following error in the web browser. **Failed really badly.**
> 
> `44440 File(s) uploadedError: INSERT INTO `lab_reports_table`(`Course_ID`, `Posted_Date`, `Deadline`, `Instructions`, `Title`, `Attachment_link_1`, `Attachment_link_2`, `Attachment_link_3`, `Attachment_link_4`,Marks,Type) VALUES ('.26.',' 2020-04-13 20:35','2020-04-30 23:59','None.','Quiz 5','','','','',2,'Individual')`
> 
> **Hui**
> 
> ![PostNewAssignment](https://user-images.githubusercontent.com/62275785/79175308-f6f47500-7e2f-11ea-8c9a-855b35c30896.jpg)
> 
> ![ErrorSQLStatement](https://user-images.githubusercontent.com/62275785/79175334-0673be00-7e30-11ea-89b7-ad6a8a5ec2da.jpg)

@spm2020spring 
Good morning sir, may i have the link to the error where the regression test failure is reported.
Considering the regression test, i realized again that in the student side under the column of Marked Submissions its now retrieving all marked submissions for all the courses that we have attended before which is weird and i guess this can be easily solved and i schedule to do that later today.

For the new error in the Teacher's Side i'm surprised since on my local server here its still working perfect. I'm able to upload new assignment and edit old assignments so i'm so sure what might be the problem but will take a look on it in the Script.php

Ashly

spm2020spring commented:
> > @Teecloudy
> > Hi Ashly
> > Thanks for the new feature that allows a lecturer to edit assignment information.
> > However, it failed a regression test, reported by Zhan Shencheng (1404325791 at qq.com). Now I am unable to post a new assignment. See the attached picture for my input.
> > Got the following error in the web browser. **Failed really badly.**
> > `44440 File(s) uploadedError: INSERT INTO `lab_reports_table`(`Course_ID`, `Posted_Date`, `Deadline`, `Instructions`, `Title`, `Attachment_link_1`, `Attachment_link_2`, `Attachment_link_3`, `Attachment_link_4`,Marks,Type) VALUES ('.26.',' 2020-04-13 20:35','2020-04-30 23:59','None.','Quiz 5','','','','',2,'Individual')`
> > **Hui**
> > ![PostNewAssignment](https://user-images.githubusercontent.com/62275785/79175308-f6f47500-7e2f-11ea-8c9a-855b35c30896.jpg)
> > ![ErrorSQLStatement](https://user-images.githubusercontent.com/62275785/79175334-0673be00-7e30-11ea-89b7-ad6a8a5ec2da.jpg)
> 
> @spm2020spring
> Good morning sir, may i have the link to the error where the regression test failure is reported.
> Considering the regression test, i realized again that in the student side under the column of Marked Submissions its now retrieving all marked submissions for all the courses that we have attended before which is weird and i guess this can be easily solved and i schedule to do that later today.
> 
> For the new error in the Teacher's Side i'm surprised since on my local server here its still working perfect. I'm able to upload new assignment and edit old assignments so i'm so sure what might be the problem but will take a look on it in the Script.php
> 
> Ashly

Thanks, Ashly. 

This following message contains the original report (in Chinese).

It says that editing an existing course was successful (see 1.png), but posting a new one has failed (see 2.png).

EditAssName.java (attached too) is the program for auto-testing.

(I have sent the attachments to your email account.)

**Hui**



-------- 转发邮件信息 --------
发件人："强化饼#" <1404325791@qq.com>
发送日期：2020-04-12 20:04:35
收件人：lanhui <lanhui@zjnu.edu.cn>
主题：回复：[毕业设计] Bug35的修复，请你测试一下
修改作业信息那边简单测试了下是成功的 -图1
但是我在新建作业那边出现了图二的问题，麻烦老师也验证下
附件有对修改作业信息用到的脚本代码，需要和之前发的登录脚本一起用

Teecloudy commented:
Good day Laoshi, i just tried again on my local server and its working properly check attached below. May you confirm that the file format and size of the attachment are right. If they are all ok then maybe we might need to withdraw the update for a while and figure out why its working on local server and not on the online server.


Ashly




------------------ Original ------------------
From:  "Software Project Management";<notifications@github.com>;
Send time: Tuesday, Apr 14, 2020 9:55 AM
To: "lanlab-org/LRR"<LRR@noreply.github.com>; 
Cc: "ASHLY"<teecloudy@qq.com>; "Mention"<mention@noreply.github.com>; 
Subject:  Re: [lanlab-org/LRR] Allowing all group member to submit assignment and allowing Lecture t… (#21)





   
@Teecloudy
 Hi Ashly
 Thanks for the new feature that allows a lecturer to edit assignment information.
 However, it failed a regression test, reported by Zhan Shencheng (1404325791 at qq.com). Now I am unable to post a new assignment. See the attached picture for my input.
 Got the following error in the web browser. Failed really badly.
 44440 File(s) uploadedError: INSERT INTO lab_reports_table(Course_ID, Posted_Date, Deadline, Instructions, Title, Attachment_link_1, Attachment_link_2, Attachment_link_3, Attachment_link_4,Marks,Type) VALUES ('.26.',' 2020-04-13 20:35','2020-04-30 23:59','None.','Quiz 5','','','','',2,'Individual')
 Hui
 
 
  
@spm2020spring
 Good morning sir, may i have the link to the error where the regression test failure is reported.
 Considering the regression test, i realized again that in the student side under the column of Marked Submissions its now retrieving all marked submissions for all the courses that we have attended before which is weird and i guess this can be easily solved and i schedule to do that later today.
 
For the new error in the Teacher's Side i'm surprised since on my local server here its still working perfect. I'm able to upload new assignment and edit old assignments so i'm so sure what might be the problem but will take a look on it in the Script.php
 
Ashly
  
Thanks, Ashly.
 
This following message contains the original report (in Chinese).
 
It says that editing an existing course was successful (see 1.png), but posting a new one has failed (see 2.png).
 
EditAssName.java (attached too) is the program for auto-testing.
 
Hui
 
-------- 转发邮件信息 --------
 发件人："强化饼#" 1404325791@qq.com
 发送日期：2020-04-12 20:04:35
 收件人：lanhui lanhui@zjnu.edu.cn
 主题：回复：[毕业设计] Bug35的修复，请你测试一下
 修改作业信息那边简单测试了下是成功的 -图1
 但是我在新建作业那边出现了图二的问题，麻烦老师也验证下
 附件有对修改作业信息用到的脚本代码，需要和之前发的登录脚本一起用
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.

spm2020spring commented:
@Teecloudy 

> Good day Laoshi, i just tried again on my local server and its working properly check attached below. May you confirm that the file format and size of the attachment are right. If they are all ok then maybe we might need to withdraw the update for a while and figure out why its working on local server and not on the online server. Ashly

I did not upload any attachment while attempting to post the new assignment.

Attachments should be optional, shouldn't they?

`4441 File(s) uploadedError: `

Why `4441 `?

Hui

Teecloudy committed:
Regression Test Failure with 4441 Upload Error

Following the previous update, the system failed a regression test and reported a 4441 error while trying to submit a new Lab report.
This was due to the condition added on line 298 in Course.php and reversing this change solved the problem.

This condition was added to allow all Group members to be updated once the Tutor Marks the Lab Report but a better method to
archieve this has been implemented on line 367 in Course.php

spm2020spring commented:
> That makes sense. Please check line 212 in Courses.php thats where the dots were added to the Course_id variable. You may try to remove them and see if it can be able to insert successfully. Compare the two images below.
> Image 1 with dots ( .$id.)
> ![a](https://user-images.githubusercontent.com/37251906/79632465-f2162500-8191-11ea-92d2-ce8bcd47856f.JPG)
> Image 2 without dots ($id)
> ![b](https://user-images.githubusercontent.com/37251906/79632470-f7736f80-8191-11ea-90a5-351298675fc0.JPG)




Thanks, @Teecloudy .  Your suggestion worked.  Please update` Courses.php`  correspondingly and I will merge this Pull Request when you are done.

**Hui**


![post-quiz5](https://user-images.githubusercontent.com/62275785/79633941-2ee61a00-819a-11ea-8c77-4b8abc201638.jpg)

Teecloudy committed:
Updating Course_id variable

In the input form of Posting New Lab assignment in COurse.php, the COurse_id variable was converted from int to a char string by adding dots around the variable. This caused a 44440 upload error since the column can only hold integers.

spm2020spring reviewed (approved):
Regression test bug fixed.  Lessons learned: ensure that data provided to the database table have the correct (expected) type.

Hui
